### PR TITLE
feat(app): Show connect alert banner on successful connection

### LINKED ISSUE #1314: Usability: Provide clear indication of successful connection
As a user, I would like to know whether my app successfully connected to my robot.

## Acceptance Criteria
- [x] If successfully connected,  display 'success' alert bar 
- [x] Alert bar is dismissable

Covered in separate ticket:
- If connection failed, display 'error' alert bar 
-- Error alert bar includes error message, if any

## Background
Upon connecting to a robot, not so clear that user is connected - no clear indication of successful connection.

## Design
- Open question: transition?
screen shot 2018-06-12 at 1 43 06 pm

### PR DESCRIPTION
**Pull request by Kadee80**
## overview
This PR closes #1314. Will open a separate issue for connectRequest error banner.

screen shot 2018-06-14 at 2 19 32 pm

## changelog

- feat(app): Show connect alert banner on successful connection

## review requests
- [x] connection banner displays on successful robot connect
- [x] connection banner is dismissible 
- [x] dismissed banner reappears on disconnect and reconnect

### CODE REVIEW

File: app/src/pages/Robots.js [hunk 1]
```diff
@@ -58,6 +59,7 @@ function RobotSettingsPage (props: Props) {
   return (
     <Page>
       <TitleBar title={robot.name} />
+      <ConnectBanner {...robot} key={Number(robot.isConnected)}/>
```
Comment (mcous) [inline app/src/pages/Robots.js]: What's the purpose of `Number` here?

File: app/src/components/RobotSettings/ConnectBanner.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  }
+
+  render () {
+    const {isConnected, name} = this.props
+    const isVisible = isConnected && !this.state.dismissed
+    const TITLE = `${name} successfully connected`
+    return (
+      <div>
+        {isVisible && (
```
Comment (mcous) [inline app/src/components/RobotSettings/ConnectBanner.js]: Could also do an early `null` return if `!isVisible`

File: app/src/pages/Robots.js (hunk 1 above)
Comment (Kadee80) [reply app/src/pages/Robots.js]: Flow was expected number or string not a bool

File: app/src/components/RobotSettings/ConnectBanner.js (hunk 2 above)
Comment (Kadee80) [reply app/src/components/RobotSettings/ConnectBanner.js]: i like that. will change in a bit

**Pull request merged**